faf-commons-data/src/test/java/com/faforever/commons/map/PreviewGeneratorFuzzTarget.java

```diff
@@ -0,0 +1,335 @@
+    "          ['INITIAL'] = GROUP {\n" +
+    "            orders = '',\n" +
+    "            platoon = '',\n" +
+    "            Units = {},\n" +
+    "          },\n" +
+    "        },\n" +
+    "      },\n" +
+    "      PlatoonBuilders = {\n" +
+    "        next_platoon_builder_id = '0',\n" +
+    "        Builders = {},\n" +
+    "      },\n" +
+    "    },\n" +
+    "    ['ARMY_9'] = {\n" +
+    "      personality = '',\n" +
+    "      plans = '',\n" +
+    "      color = 0,\n" +
+    "      faction = 0,\n" +
+    "      Economy = {mass = 0, energy = 0},\n" +
+    "      Alliances = {},\n" +
+    "      ['Units'] = GROUP {\n" +
+    "        orders = '',\n" +
+    "        platoon = '',\n" +
+    "        Units = {\n" +
+    "          ['INITIAL'] = GROUP {\n" +
+    "            orders = '',\n" +
+    "            platoon = '',\n" +
+    "            Units = {},\n" +
+    "          },\n" +
+    "        },\n" +
+    "      },\n" +
+    "      PlatoonBuilders = {\n" +
+    "        next_platoon_builder_id = '0',\n" +
+    "        Builders = {},\n" +
+    "      },\n" +
+    "    },\n" +
+    "    ['NEUTRAL_CIVILIAN'] = {\n" +
+    "      personality = '',\n" +
+    "      plans = '',\n" +
+    "      color = 0,\n" +
+    "      faction = 0,\n" +
+    "      Economy = {mass = 0, energy = 0},\n" +
+    "      Alliances = {},\n" +
+    "      ['Units'] = GROUP {\n" +
+    "        orders = '',\n" +
+    "        platoon = '',\n" +
+    "        Units = {\n" +
+    "          ['INITIAL'] = GROUP {\n" +
+    "            orders = '',\n" +
+    "            platoon = '',\n" +
+    "            Units = {},\n" +
+    "          },\n" +
+    "        },\n" +
+    "      },\n" +
+    "      PlatoonBuilders = {\n" +
+    "        next_platoon_builder_id = '0',\n" +
+    "        Builders = {},\n" +
+    "      },\n" +
+    "    },\n" +
+    "  },\n" +
+    "}\n";
+
+    public static void fuzzerTestOneInput(FuzzedDataProvider data) throws IOException {
+      Path currentDir = Path.of(".");
+      Path saveLua = currentDir.resolve("example_save.lua");
+      Path scmap = currentDir.resolve("example.scmap");
+      float randomFloat = data.consumeFloat();
+      if(randomFloat == Float.NEGATIVE_INFINITY || randomFloat == Float.POSITIVE_INFINITY || Float.isNaN(randomFloat)){
+        return;
+      }
+      Files.writeString(saveLua, String.format(saveLuaTemplate, randomFloat, randomFloat, randomFloat), StandardOpenOption.CREATE,
+        StandardOpenOption.TRUNCATE_EXISTING);
+      Files.write(scmap, data.consumeRemainingAsBytes(), StandardOpenOption.CREATE,
+        StandardOpenOption.TRUNCATE_EXISTING);
+
+      try {
+        PreviewGenerator.generatePreview(
+          Path.of("."), 256, 256
+        );
+        /*try{
+          ImageIO.write(bufferedImage, "png", Path.of("./"+Math.random()+".png").toFile());
+        }catch (Exception e){
+          throw new Error(e);
+        }*/
+      } catch (PreviewGenerator.IllegalInput ignore) {
```

Comment:
Are these commented lines really needed?

Reply:
It is just for fun to see what the generated images look like

Reply:
I think that JavaDoc is a better place for this.